tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -561,6 +563,34 @@ public function testDateArithmetics()
         self::assertEquals('2004-01-01', date('Y-m-d', strtotime($row['sub_years'])), "Subtracting years should end up on 2004-01-01");
     }
 
+    public function testSqliteDateArithmeticWithDynamicInterval()
+    {
+        if (! $this->_conn->getDriver() instanceof PDOSqliteDriver) {
```

Comment:
One last thing from me: please check platform instead of the driver since it's a platform-specific feature, not driver-specific.

Reply:
Good spot, I've made that change now.